components/calendar/calendar.js

```diff
@@ -0,0 +1,480 @@
+		dates.push(week);
+	}
+
+	return dates;
+}
+
+export function getNextMonth(month) {
+	return (month === 11) ? 0 : (month + 1);
+}
+
+export function getNumberOfDaysFromPrevMonthToShow(month, year) {
+	const firstDayOfMonth = new Date(year, month, 1).getDay();
+	let numDaysFromLastMonthToShowThisMonth = 0;
+	if (firstDayOfMonth !== firstDayOfWeek) {
+		numDaysFromLastMonthToShowThisMonth = firstDayOfMonth - firstDayOfWeek;
+		if (numDaysFromLastMonthToShowThisMonth < 0) {
+			numDaysFromLastMonthToShowThisMonth += daysInWeek;
+		}
+	}
+	return numDaysFromLastMonthToShowThisMonth;
+}
+
+export function getNumberOfDaysInMonth(month, year) {
+	return new Date(year, month + 1, 0).getDate();
+}
+
+export function getNumberOfDaysToSameWeekPrevMonth(month, year) {
+	const prevMonth = getPrevMonth(month);
+	const prevMonthYear = prevMonth === 11 ? year - 1 : year;
+	const numDaysPrevMonth = getNumberOfDaysInMonth(prevMonth, prevMonthYear);
+	const numDaysFromPrevPrevMonthToShowPrevMonth = getNumberOfDaysFromPrevMonthToShow(prevMonth, prevMonthYear);
+	const numWeeksPrevMonth = Math.ceil((numDaysFromPrevPrevMonthToShowPrevMonth + numDaysPrevMonth) / 7);
+
+	const firstDayOfMonth = new Date(year, month, 1).getDay();
+	const hasDaysFromPrevMonth = (firstDayOfMonth !== firstDayOfWeek);
+	return daysInWeek * (numWeeksPrevMonth - ((hasDaysFromPrevMonth) ? 1 : 0));
+}
+
+export function getPrevMonth(month) {
+	return (month === 0) ? 11 : (month - 1);
+}
+
+export function parseDate(val) {
+	if (!val) return null;
+
+	let date = val;
+	if (val.includes('T')) {
+		const dateTimeParts = val.split('T');
+		date = dateTimeParts[0];
+	}
+	const dateParts = date.split('-');
+	if (dateParts.length !== 3) {
+		throw new Error('Invalid selected-value date input: Expected format is YYYY-MM-DDTHH:mm:ss.sssZ or YYYY-MM-DD');
+	}
+
+	return new Date(dateParts[0], parseInt(dateParts[1]) - 1, dateParts[2]);
+}
+
+class Calendar extends LocalizeStaticMixin(LitElement) {
+
+	static get properties() {
+		return {
+			selectedValue: { type: String, attribute: 'selected-value' },
+			summary: { type: String },
+			_focusDate: { type: Object },
+			_shownMonth: { type: Number }
+		};
+	}
+
+	static get styles() {
+		return [calendarStyles, bodySmallStyles, heading4Styles];
+	}
+
+	static get resources() {
+		return {
+			'ar': { show: 'Show {month} ' },
+			'en': { show: 'Show {month}' },
+			'es': { show: 'Show {month} ' },
+			'fr': { show: 'Show {month} ' },
+			'ja': { show: 'Show {month} ' },
+			'ko': { show: 'Show {month} ' },
+			'nl': { show: 'Show {month} ' },
+			'pt': { show: 'Show {month} ' },
+			'sv': { show: 'Show {month} ' },
+			'tr': { show: 'Show {month} ' },
+			'zh': { show: 'Show {month} ' },
+			'zh-tw': { show: 'Show {month} ' }
+		};
+	}
+
+	constructor() {
+		super();
+
+		this._labelId = getUniqueId();
+	}
+
+	firstUpdated(changedProperties) {
+		super.firstUpdated(changedProperties);
+
+		this._today = new Date();
```

Comment:
This could be a potential source of bugs as this will be in the user computer local timezone. Not sure what this should be... UTC "now" then converted to local?